crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,409 @@
+            can_be_sorted: !contains_shorthand_after_longhand(&items)
+                && !contains_unknown_property(&items),
+            first_out_of_order_pair,
+        })
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        if !state.can_be_sorted || state.first_out_of_order_pair.is_none() {
+            return None;
+        }
+
+        if let Some((a, b)) = &state.first_out_of_order_pair {
+            let a_description = node_short_description(a);
+            let b_description = node_short_description(b);
+            return Some(RuleDiagnostic::new(
+                rule_category!(),
+                state.block.range(),
+                markup! {
+                    "Properties should be sorted: "<Emphasis>{ a_description }</Emphasis>" should be before "<Emphasis>{ b_description }</Emphasis>"."
+                },
+            ));
+        }
+
+        None
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<CssRuleAction> {
+        if !state.can_be_sorted || state.first_out_of_order_pair.is_none() {
+            return None;
+        }
+
+        let mut mutation = ctx.root().begin();
+
+        let original_items = to_block_items_with_position(&state.block);
+        let mut sorted_items = original_items.clone();
+        sorted_items.sort_by(|a, b| sort_info(&a.node).cmp(&sort_info(&b.node)));
+
+        let pairs = original_items.iter().zip(sorted_items.iter());
+
+        for (item_to_replace, replacement) in pairs {
+            if item_to_replace.position != replacement.position {
+                mutation.replace_node_discard_trivia(
+                    item_to_replace.node.clone(),
+                    replacement.node.clone(),
+                );
+            }
+        }
+
+        return Some(RuleAction::<CssLanguage>::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { "Sort these properties" }.to_owned(),
+            mutation,
+        ));
+    }
+}
+
+#[derive(Clone)]
+struct NodeWithPosition {
+    position: usize,
+    node: AnyCssDeclarationOrRule,
+}
+
+fn to_block_items_with_position(block: &CssDeclarationOrRuleBlock) -> Vec<NodeWithPosition> {
+    block
+        .items()
+        .into_iter()
+        .enumerate()
+        .map(|(position, item)| NodeWithPosition {
+            position,
+            node: item,
+        })
+        .collect()
+}
+
+// takes the unsorted list and returns the first item that is out of order, and the item it should be placed before
+// returns None if the list is sorted
+fn get_first_out_of_order_pair(
+    items: &[NodeWithPosition],
+) -> Option<(NodeWithPosition, NodeWithPosition)> {
+    if items.is_empty() {
+        return None;
+    }
+
+    let mut first_out_of_order_node: Option<NodeWithPosition> = None;
+
+    // find the first node that's 'less than' the previous node
+    let mut prev_item_info = sort_info(&items[0].node);
```

Comment:
Don't access items of lists using indexes, unless you are 100% sure it's safe. Rust exposes safe methods to work around this.

This means that you can simplify your code like this:

```rs
let first_item = items.get(0)?;
// notice the try operator ?, which returns early if `get()` returns `None`
let mut prev_item_info = sort_info(&first_item.node);
```

And you don't need the `is_empty` check anymore